File: app/src/main/java/com/mohsenoid/rickandmorty/view/character/list/CharacterListContract.java
package com.mohsenoid.rickandmorty.view.character.list;

import com.mohsenoid.rickandmorty.domain.entity.CharacterEntity;
import com.mohsenoid.rickandmorty.view.base.BasePresenter;
import com.mohsenoid.rickandmorty.view.base.BaseView;

import java.util.List;

public interface CharacterListContract {

    interface View extends BaseView {

        void onNoOfflineData();

        void setCharacters(List<CharacterEntity> characters);

        void updateCharacter(CharacterEntity character);
    }

    interface Presenter extends BasePresenter<View> {

        void setCharacterIds(List<Integer> characterIds);

        void loadCharacters();

        void killCharacter(CharacterEntity character);
    }
}

File: app/src/main/kotlin/com/mohsenoid/rickandmorty/view/character/list/CharacterListContract.kt
package com.mohsenoid.rickandmorty.view.character.list

import com.mohsenoid.rickandmorty.domain.model.ModelCharacter
import com.mohsenoid.rickandmorty.view.base.BasePresenter
import com.mohsenoid.rickandmorty.view.base.BaseView

interface CharacterListContract {

    interface View : BaseView {

        fun onNoOfflineData()

        fun setCharacters(characters: List<ModelCharacter>)

        fun updateCharacter(character: ModelCharacter)
    }

    abstract class Presenter : BasePresenter<View>() {

        abstract suspend fun loadCharacters(characterIds: List<Int>)

        abstract suspend fun killCharacter(character: ModelCharacter)
    }
}
